decentraland/screens/ListLandForSaleScreen.js

```diff
@@ -4,78 +4,60 @@ import { setLandForSaleList, selectLandToBuy } from "../redux/actions";
-      const landForSaleList = await this._getLandForSaleList();
+      showInfo("Loading assets for sale...");
+      const landForSaleList = await this._getAssetsForSale();
       setLandForSaleList(landForSaleList);
     } catch (err) {
-      console.error(err);
+      showError(err);
     }
   };
 
-  // Note: This function is assuming that:
-  // - All estates have a sell order
-  // - The total supply of estates is small
-  // TODO: Rewrite this function when we move to testnet
-  _getLandForSaleList = async () => {
-    const orders = [];
-    const totalSupply = await this.estateContract.totalSupply();
-
-    for (let estateId = 1; estateId <= Number(totalSupply); estateId++) {
-      const order = this._getLandForSale(estateId);
-      orders.push(order);
-    }
+  _getAssetsForSale = async () => {
+    const decentralandUtils = new DecentralandUtils();
+    const { getOpenSellOrders } = decentralandUtils;
 
-    if (orders.length == 0) console.warn("There isn't any land for sale right now.");
+    const fromBlock = 0;
+    const openSellOrdersEvents = await getOpenSellOrders(fromBlock);
+    const assetsForSale = [];
 
-    return await Promise.all(orders);
+    for (let event of openSellOrdersEvents) {
+      let { values: order } = event;
+      let assetForSale = await this._prepareAssetForSale(order);
+      assetsForSale.push(assetForSale);
+    }
+    return assetsForSale;
   };
 
-  _getLandForSale = async estateId => {
-    const estateName = await this.estateContract.getMetadata(estateId);
-    const [
-      orderId,
-      seller,
-      price,
-      expiresAt,
-    ] = await this.marketplaceContract.auctionByAssetId(estateId);
-
-    const hasOrder = parseInt(orderId, 16) !== 0;
-    if (!hasOrder) throw Error(`Estate (id:${estateId}) has no sell order.`);
+  _prepareAssetForSale = async assetForSale => {
+    const { nftAddress } = assetForSale;
+    let contracts = getContracts();
+    const { estateContract, landContract } = contracts;
 
-    // Note: Conversion to USD will be implemented on v0.2.0
-    const manaPerUsd = 30;
-    const priceMana = Number(price.toString()) / 1e18;
-    const priceUSD = Number(priceMana / manaPerUsd).toFixed(2);
-    const imgUrl = `https://api.decentraland.org/v1/estates/${estateId}/map.png`;
+    const isParcel = addressesAreEqual(nftAddress, landContract.getAddress());
+    const isEstate = addressesAreEqual(nftAddress, estateContract.getAddress());
 
-    return {
-      id: orderId,
-      priceMana,
-      priceUSD,
-      seller,
-      expiresAt,
-      // TODO: Create an enum type for identify asset
-      asset: {
-        id: estateId,
-        name: estateName,
-        img: imgUrl,
-      },
-    };
+    if (isEstate) {
+      return await prepareEstateForSale(estateContract, assetForSale);
+    } else if (isParcel) {
+      return await prepareParcelForSale(landContract, assetForSale);
+    } else {
+      throw new Error(`The asset should be a Parcel or an Estate.`);
```

Comment:
I'm not sure it makes sense to capitalize parcel, given that what it's really called it land and parcel is just the word used for discrete pieces of land

Reply:
```suggestion
      throw new Error(`The asset should be a parcel of land or an estate.`);
```

Reply:
Done.

Reply:
Thanks! Resolving